scripts/installer.sh

```diff
@@ -0,0 +1,282 @@
+  if ! tar -xzf "${TEMP_DIR}/${ARCHIVE_NAME}.tar.gz" -C "${TEMP_DIR}" >/dev/null; then
+    print_error "Failed to unarchive Cloudlift package."
+  fi
+}
+
+install_cloudlift() {
+  print_info "Installing Cloudlift package..."
+  [ -d "${PACKAGE_DEST_DIR}/${PACKAGE_NAME}" ] && rm -rf "${PACKAGE_DEST_DIR}/${PACKAGE_NAME}"
+  [ -L "${PACKAGE_BIN_DIR}/${PACKAGE_NAME}" ] && rm "${PACKAGE_BIN_DIR}/${PACKAGE_NAME}"
+
+  if ! cp -r "${TEMP_DIR}/${ARCHIVE_NAME}" "${PACKAGE_DEST_DIR}/${PACKAGE_NAME}"; then
+    print_error "Failed to copy Cloudlift package."
+  fi
+  if ! ln -s "${PACKAGE_DEST_DIR}/${PACKAGE_NAME}/cloudlift" "${PACKAGE_BIN_DIR}/${PACKAGE_NAME}"; then
+    print_error "Failed to create symlink."
+  fi
+}
+
+test_installation() {
+  print_info "Testing Cloudlift installation..."
+  if ! command -v "cloudlift" >/dev/null; then
+    print_error "Failed to install Cloudlift package."
+  fi
+  if ! cloudlift --help >/dev/null; then
+    print_error "Failed to run Cloudlift package."
+  fi
+  installed_version=$(cloudlift --version | cut -d' ' -f3)
+  if [ "${installed_version}" != "${PACKAGE_VERSION}" ]; then
+    print_error "Cloudlift package version mismatch."
+  fi
+  print_info "Cloudlift package installation test passed."
+}
+
+uninstall_cloudlift() {
+  print_info "Uninstalling Cloudlift package..."
+  if ! command -v "cloudlift" >/dev/null; then
+    print_info "Cloudlift package not found. Exiting..."
+    exit 0
+  fi
+  current_version=$(cloudlift --version | cut -d' ' -f3)
+  print_info "Current Cloudlift package version: ${current_version}"
+  confirm_message "Do you want to uninstall Cloudlift package?"
+
+  [ -d "${PACKAGE_DEST_DIR}/${PACKAGE_NAME}" ] && rm -rf "${PACKAGE_DEST_DIR}/${PACKAGE_NAME}"
+  [ -L "${PACKAGE_BIN_DIR}/${PACKAGE_NAME}" ] && rm "${PACKAGE_BIN_DIR}/${PACKAGE_NAME}"
+
+  if command -v "cloudlift" >/dev/null; then
+    print_error "Failed to uninstall Cloudlift package."
+  fi
+  print_info "Cloudlift package uninstalled successfully."
+}
+
+preflight_check() {
+  print_info "Running preflight checks..."
+  [ -z "${TERM}" ] && AUTO_APPROVE=true
+  for cmd in curl tar gzip; do
+    if ! command -v "$cmd" >/dev/null; then
+      print_error "$cmd is required."
+    fi
+  done
+
+  [ "${PACKAGE_VERSION}" = "latest" ] && get_latest_release
+
+  if command -v "cloudlift" >/dev/null; then
+    current_version=$(cloudlift --version | cut -d' ' -f3)
+    [ -z "${current_version}" ] && print_warning "The current Cloudlift did not install properly or is corrupted."
+    print_info "Cloudlift package already installed; version: v${current_version}"
+    confirm_message "Do you want to reinstall Cloudlift with version v${PACKAGE_VERSION}?"
+  fi
+}
+
+# ----------------------------------------
+# Main Execution
+# ----------------------------------------
+
+parse_arguments() {
+  while [ $# -gt 0 ]; do
+    case "$1" in
+    -h | --help)
+      usage
+      exit 0
+      ;;
```

Comment:
there's no function named `usage` in the installer script

Reply:
my bad. the function `usage` got dropped while moving the installer script. it has been updated.